packages/happy-dom/src/query-selector/SelectorItem.ts

```diff
@@ -8,7 +8,7 @@ const ATTRIBUTE_NAME_REGEXP = /[^a-zA-Z0-9-_$]/;
 const PSUEDO_REGEXP = /:([a-zA-Z-]+)\(([0-9n+-]+|odd|even)\)|:not\(([^)]+)\)|:([a-zA-Z-]+)/g;
 const CLASS_REGEXP = /\.([a-zA-Z0-9-_$]+)/g;
 const TAG_NAME_REGEXP = /^[a-zA-Z0-9-]+/;
-const ID_REGEXP = /#[A-Za-z][-A-Za-z0-9_]*/g;
+const ID_REGEXP = /#[^\s]+/g;
```

Comment:
```suggestion
const ID_REGEXP = /#[A-Za-z][A-Za-z0-9_-]*/g;
```

Based on what is written on [this Mozilla documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id), it seems like browsers will throw an error for any other value than the above RegExp.

Example:
```javascript
// Throws an error
document.querySelector('#1myid');

// Throws an error
document.querySelector('#myid[');
```